test/cli/whole-program_test.py

```diff
@@ -11,15 +11,17 @@
 # TODO: test CheckBufferOverrun
 
 
-def __create_compile_commands(dir, entries):
+def __create_compile_commands(dir, filePaths, definesList=[[None]]):
     j = []
-    for e in entries:
-        f = os.path.basename(e)
-        obj = {
-            'directory': os.path.dirname(os.path.abspath(e)),
-            'command': 'gcc -c {}'.format(f),
-            'file': f
-        }
+    for i, fp in enumerate(filePaths):
+        f = os.path.basename(fp)
+
+        for defines in (definesList[i] if len(definesList) > i else [None]):
+            obj = {
```

Comment:
I doubt that in a real world project a compile commands will contain multiple entries for the same file compiled with different flags.

Reply:
That actually exists in our own project. We compile the same source files within different scopes. There also can be multiple commands within a single entry I think. We have a workaround for that and it might be an extension outside of the actual "documentation" (there is no "real" specification). Need to look that up again. We might also need to document and make things opt-in.

This also plays into the de-duplication of stuff which is still not complete.

Reply:
ok

Reply:
> There also can be multiple commands within a single entry I think

I doubt that. But arguments can be provided in an array instead of a string maybe that's what you're thinking about.

So if it is allowed with multiple entries for the same file in a compile commands then it's a good thing to test this explicitly. But I am still not sure if I like this test if that is explicitly what you want to test. I wonder if we can write more specific tests in testrunner that checks that the file is loaded correctly etc.

Reply:
Cannot remember - I will look it up later and file a ticket.